### PR TITLE
TenantDetail and TenantDelete handlers

### PR DESCRIPTION
**Pull request by daniellemaxwell**
### Scope of changes

Adds server-side handlers for TenantDetail and TenantDelete along with corresponding tests. 
`TenantDetail` will allow a user to pass in a given tenant id and get a summary detail for the specified tenant.
`TenantDelete` will allow a user to pass in a given tenant id and delete a tenant.

The complexity for this PR is low. Comments to the `TenantCreate` handler and test have been added as well.

Fixes SC-10278

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [x] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

Describe how reviewers can test this change to be sure that it works correctly. Add a checklist if possible.

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [x]  I have added new test fixtures as needed to support added tests
- [ ]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.

### CODE REVIEW

File: pkg/tenant/tenants.go
Comment (daniellemaxwell) [inline pkg/tenant/tenants.go:13]: This handler will be implemented via a separate PR.

File: pkg/tenant/tenants.go
Comment (daniellemaxwell) [inline pkg/tenant/tenants.go:137]: This handler will be implemented via a separate handler and this was only a position change.

File: pkg/tenant/tenants_test.go
Comment (pdeziel) [inline pkg/tenant/tenants_test.go:54]: Can we change this to a `require.NoError`? I think as the test is written currently, if we get an error here we will just return out of the test and the rest of the test code is skipped, which makes it look like the test has passed.

File: pkg/tenant/tenants_test.go
Comment (pdeziel) [inline pkg/tenant/tenants_test.go:72]: Can we add a few quick cases here and in `TestTenantDelete` to cover the error cases? We just need to be sure that if the required params are not supplied we're getting the correct HTTP status response back with an error. We should also test that 404s are returned when the trtl mock is set up to return an error.

**Review by pdeziel — DISMISSED** [review]
@daniellemaxwell This is a great start and the tenant API definitely makes sense at a high level. I did have some comments about the HTTP status codes we are returning but hopefully those are quick fixes. We probably also need to add a bit more test coverage as I've indicated in the comments to assert that we are returning the correct responses. Let me know if you want to discuss!

**Review by pdeziel — COMMENTED** [review]
Thanks for making the changes! I'll let @bbengfort have a chance to look at this before merging.

File: pkg/tenant/tenants.go
Comment (bbengfort) [inline pkg/tenant/tenants.go:21]: I like the doc strings :-)

File: pkg/tenant/tenants.go
Comment (bbengfort) [inline pkg/tenant/tenants.go:41]:
```suggestion
		c.JSON(http.StatusBadRequest, api.ErrorResponse("tenant id cannot be specified on create"))
```

File: pkg/tenant/tenants.go
Comment (bbengfort) [inline pkg/tenant/tenants.go:103]:
```suggestion
```

Stray debugging line!

**Review by bbengfort — APPROVED** [review]
Looks great - thank you @pdeziel for the initial review and thank you @daniellemaxwell for sticking with the PR!

**Pull request merged**
